crates/runtime/src/accelerated_table.rs

```diff
@@ -372,16 +371,44 @@ impl AcceleratedTable {
                     }
                 }
                 AccelerationRefreshMode::Full(receiver) => {
+                    let filter_converter = match (refresh.period, refresh.time_column) {
+                        (None, Some(_) | None) => {
+                            None
+                        },
+                        (Some(_), None) => {
+                            tracing::warn!("[refresh] No time_column is provided, refresh_period will be ignored");
+                            None
+                        }
+                        (Some(_), Some(column)) => {
+                            let schema = federated.schema();
+                            let field = schema.column_with_name(column.as_str());
+                            TimestampFilterConvert::create(field, Some(column), refresh.time_format)
+                        }
+                    };
+
                     let mut refresh_stream = ReceiverStream::new(receiver);
 
                     while refresh_stream.next().await.is_some() {
-                        tracing::info!("Refreshing data for {dataset_name}");
+                        tracing::info!("[refresh] Refreshing data for {dataset_name}");
                         status::update_dataset(&dataset_name, status::ComponentStatus::Refreshing);
                         let timer = TimeMeasurement::new("load_dataset_duration_ms", vec![("dataset", dataset_name.clone())]);
-                        let all_data = match get_all_data(&mut ctx, OwnedTableReference::bare(dataset_name.clone()), Arc::clone(&federated), refresh_sql.clone()).await {
+                        let  filters = match (refresh.period, filter_converter.clone()){
+                            (Some(period), Some(converter)) => {
+                                let start = SystemTime::now() - period;
+
+                                let Ok(timestamp) = get_timestamp(start) else {
+                                    tracing::error!("[refresh] Failed to get timestamp");
+                                    continue;
+                                };
+                                vec![converter.convert(timestamp, Operator::Gt)]
+                            },
+                            _ => vec![],
+                        };
+
+                        let all_data = match get_all_data(&mut ctx, OwnedTableReference::bare(dataset_name.clone()), Arc::clone(&federated), refresh.sql.clone(), filters).await {
```

Comment:
```suggestion
                        let data = match get_data(&mut ctx, OwnedTableReference::bare(dataset_name.clone()), Arc::clone(&federated), refresh.sql.clone(), filters).await {
```

This isn't technically all data anymore.

Reply:
good point.